src/main/scala/mesosphere/marathon/Main.scala

```diff
@@ -29,12 +29,30 @@ class MarathonApp extends App {
     )
   }
 
-  override val conf = new AllConf(args)
+  private val envPrefix = "MARATHON_"
+  private val envArgs: Array[String] = {
```

Comment:
Prefer `Seq[]` please. You should be able to do something like

`
sys.env.withFilter(_._1.startsWith(EnvPrefix)).flatMap { case (key, value) => Seq(s"--${key.replaceAll(EnvPrefix, "").toLowerCase.trim}", if (value.trim.length > 0) value else Nil) }(collection.breakOut)
`